src/core/middleware/store.ts

```diff
@@ -0,0 +1,54 @@
+import { destroy, invalidator, create } from '../vdom';
+import injector from '../middleware/injector';
+import Store, { StatePaths, Path } from '../../stores/Store';
+import { Process } from '../../stores/process';
+
+const factory = create({ destroy, invalidator, injector });
+
+export const createStoreMiddleware = <S = any>(initial?: (store: Store<S>) => void) => {
```

Comment:
Accepts an optional `initial` (perhaps should be called `init` or `initialize`) function that receives the factorie'd store and can initialize state etc.